components/Input/InputTextFrom.vue

```diff
@@ -0,0 +1,84 @@
+
+<script setup lang="ts">
+import { ref } from 'vue'
+import type { Ref } from 'vue'
+
+const props = defineProps({
+  className: {
+    type: [String],
+  },
+  type: {
+    type: [String],
+  },
+  placeholder :{
+    type: [String]      
+  },
+  icon :{
+    type: [Boolean]
+  },
+  src :{
+    type :[String]
+  },
+
+})
+
+let type: Ref<String> = ref('text')
+
+type.value = props.type
+console.log(type.value)
+
+function showPassword() {
+   if (type.value === "password") {
+    type.value = "text";
+  } else {
+    type.value = "password";
+  }
+}
+
+
+
+</script>
+<template>
+    <div class="relative wrapper-input-text">
+      <img :src="props.src" v-if="props.icon === true" class="img-input-text absolute right-3 z-50 p-1" alt="" @click="showPassword">
+      <input :type="type" placeholder=" " :class="'input-text outline-none border border-gray-150 w-full bg-none p-4  font-poppins text-sm absolute top-0 left-0 h-full '+props.className" />
```

Comment:
please add support for `v-model`

you can follow these example:

Component:
```vue
<script>
defineProps({
  modelValue: {
    type: [String],
  },
})
const emit = defineEmits(['update:modelValue'])

function updateValue(event){
   emit('update:modelValue',event.target.value)
}
</script>
<template>
<input :value="modelValue" @update="updateValue"/>
</template>

```
Usage:
```vue
<InputTextForm v-model="your ref"/>
```